src/fondant/executor.py

```diff
@@ -195,12 +207,18 @@ def execute(self, component_cls: t.Type[Component]) -> None:
         """
         input_manifest = self._load_or_create_manifest()
 
-        component = component_cls(self.spec, **self.user_arguments)
-        output_df = self._execute_component(component, manifest=input_manifest)
+        output_df = None
+        if self.execute_component:
+            logging.info("Executing component")
+            component = component_cls(self.spec, **self.user_arguments)
+            output_df = self._execute_component(component, manifest=input_manifest)
+        else:
+            logging.info("Cached component run. Skipping component execution")
 
         output_manifest = input_manifest.evolve(component_spec=self.spec)
 
-        self._write_data(dataframe=output_df, manifest=output_manifest)
+        if self.execute_component:
```

Comment:
So a skipped component still technically runs it just uploads a manifest.

Reply:
yeah indeed, the `execution` here refers to loading/transforming the data. Perhaps we need a better name